src/provider/rainbowKit/RainbowKitAccount.ts

```diff
@@ -0,0 +1,378 @@
+import { AddressUtils, AminoTypesMap, CarbonSDK, CarbonTx, EVMChain as EVMChainV2, Models } from "@carbon-sdk/index";
+import { ETH_SECP256K1_TYPE, parseChainId, populateEvmTransactionDetails } from "@carbon-sdk/util/ethermint";
+import { CarbonSigner, CarbonSignerTypes } from "@carbon-sdk/wallet";
+import { Algo, EncodeObject } from "@cosmjs/proto-signing";
+import { ethers } from "ethers";
+import { parseEvmError } from "../metamask/error";
+import { DirectSignResponse } from "@keplr-wallet/types";
+import { AuthInfo, TxBody } from "cosmjs-types/cosmos/tx/v1beta1/tx";
+import { TxTypes, registry } from "@carbon-sdk/codec";
+import { StdFee } from "@cosmjs/stargate";
+import { SWTHAddressOptions } from "@carbon-sdk/util/address";
+import { AminoMsg, makeSignDoc } from "@cosmjs/amino";
+import { constructEIP712Tx } from "@carbon-sdk/util/eip712";
+import { signTransactionWrapper } from "@carbon-sdk/util/provider";
+import { legacyConstructEIP712Tx } from "@carbon-sdk/util/legacyEIP712";
+import { carbonNetworkFromChainId } from "@carbon-sdk/util/network";
+import { Blockchain, BlockchainV2, getBlockchainFromChainV2 } from "@carbon-sdk/util/blockchain";
+import { CarbonEvmChainIDs, Network } from "@carbon-sdk/constant";
+import { ARBITRUM_MAINNET, ARBITRUM_TESTNET, BSC_MAINNET, BSC_TESTNET, CARBON_EVM_DEVNET, CARBON_EVM_LOCALHOST, CARBON_EVM_MAINNET, CARBON_EVM_TESTNET, ETH_MAINNET, ETH_TESTNET, ChangeNetworkParam, OKC_MAINNET, OKC_TESTNET, POLYGON_MAINNET, POLYGON_TESTNET } from "../../constant";
+
+export interface RainbowKitWalletOpts {
+  publicKeyMessage?: string
+  publicKeyBase64: string;
+  walletProvider: string;
+}
+
+interface RequestArguments {
+  method: string;
+  params?: unknown[] | object;
+}
+
+interface RainbowkitAPI {
+  chainId: string | null;
+  request: (args: RequestArguments) => Promise<unknown>;
+  on: (eventName: string, listener: (...args: unknown[]) => void) => any;
+}
+
+export interface SyncResult  {
+  blockchain?: Blockchain | BlockchainV2;
+  chainId?: number;
+}
+
+export type EVMChain = EVMChainV2;
```

Comment:
Think this is defined in `src/provider/metamask/MetaMask.ts` also. Since this type seems very generic, maybe move this to a `src/constant` file and import here (do the same for MetaMask.ts).